src/ads/ads-controller.js

```diff
@@ -19,19 +18,18 @@ class AdsController extends FakeEventTarget implements IAdsController {
    * @param {IAdsController} adsPluginController - the controller of the current plugin instance.
    */
   constructor(player: Player, adsPluginController: IAdsController) {
-    super();
     this._player = player;
     this._adsPluginController = adsPluginController;
     this._initMembers();
     this._addBindings();
   }
 
   /**
-   * Getter to the no more ads to play indication.
-   * @returns {boolean} - Whether all ads completed.
+   * Getter whether there is a post roll
+   * @returns {boolean} - there is a post roll.
    */
-  get allAdsCompleted(): boolean {
-    return this._allAdsCompleted;
+  get hasPostRoll(): boolean {
```

Comment:
Dont forget this class is exposed via player.ads. thia API makes no sense to expose. App can get that information via the proper event payload.

Reply:
In addition, 
1. This is not a common ads API.
2. You broke ads API which is implemented also in the cast sender!

Reply:
Ok, I tend to agree with @dan-ziv on this.
The api should remain allAdsCompletes and you can manage it internally by checking if there’s a postroll and if so wait for playback of post roll to update the status

Reply:
@OrenMe @dan-ziv  I can't remain it as `allAdsCompletes` because now it returns `true` before `ALL_ADS_COMPLETED` event. consider case of preroll/midroll only.

Reply:
@dan-ziv regarding the sender, i thought about this, but you anyway should your code, else you have same bug in case of non-linear ad.

Reply:
@yairans still this API makes no real sense.
You can manage this internally.
allAdsCompletes can be set to true also in case of midroll cause after midroll it will be true.
The only thing this flag actually says is that there are no more ads.